src/runtime_src/tools/xclbinutil/SectionIPLayout.cxx

```diff
@@ -119,6 +119,75 @@ SectionIPLayout::getIPControlType(std::string& _sIPControlType) const
   throw std::runtime_error(errMsg);
 }
 
+const std::string
+SectionIPLayout::getFunctionalStr(PS_FUNCTIONAL eFunctional) const
+{
+  switch (eFunctional) {
+    case FC_DPU:
+      return "DPU";
+    case FC_PREPOST:
+      return "PrePost";
+    default:
+      return (boost::format("UNKNOWN (%d)") % static_cast<unsigned int>(eFunctional)).str();
+  }
+}
+
+PS_FUNCTIONAL
+SectionIPLayout::getFunctionalNoError(std::string& sFunctional)
+{
+  if (sFunctional == "DPU") 
+    return FC_DPU;
+  else if (sFunctional == "PrePost") 
+    return FC_PREPOST;
+  else
+    return FC_UNKNOWN;
+}
+
+PS_FUNCTIONAL
+SectionIPLayout::getFunctional(std::string& sFunctional) const
+{
+  PS_FUNCTIONAL eFunctional = getFunctionalNoError(sFunctional); 
+  if (eFunctional == FC_UNKNOWN) {
+    const std::string errMsg = "ERROR: Unknown Functional: '" + sFunctional + "'";
+    throw std::runtime_error(errMsg);
+  }
+  return eFunctional;
+}
+
+const std::string
+SectionIPLayout::getSubTypeStr(PS_SUBTYPE eSubType) const
+{
+  switch (eSubType) {
+    case ST_PS:
+      return "PS";
+    case ST_DPU:
+      return "DPU";
+    default:
+      return (boost::format("UNKNOWN (%d)") % static_cast<unsigned int>(eSubType)).str();
+  }
+}
+
+PS_SUBTYPE
+SectionIPLayout::getSubTypeNoError(std::string& sSubType)
+{
+  if (sSubType == "PS")
+    return ST_PS;
+  else if (sSubType == "DPU")
+    return ST_DPU;
+  else 
+    return ST_UNKNOWN;
+}
+
+PS_SUBTYPE
+SectionIPLayout::getSubType(std::string& sSubType) const
+{
+  PS_SUBTYPE eSubType = getSubTypeNoError(sSubType); 
```

Comment:
See earlier comments about try/catch blocks and throwing an exception.

Reply:
code refactored